utils/comparable_bytes.cc

```diff
@@ -131,6 +131,31 @@ static void decode_signed_long_type(managed_bytes_view& src, bytes_ostream& out)
     out.write(src, bytes_to_read);
 }
 
+// Fixed length signed floating point number encode/decode.
+// To encode :
+//   If positive : invert first bit to make it greater than all negatives
+//   If negative : invert all bytes to make negatives with bigger magnitude smaller
+// Decoding is identical except the logic to identify postive/negative values
+template <std::floating_point T>
+static void convert_fixed_length_float(managed_bytes_view& src, bytes_ostream& out, bool perform_encode) {
+    // Read and write float as uint32_t; double as uint64_t;
+    using uint_t = std::conditional_t<std::is_same_v<T, float>, uint32_t, uint64_t>;
+
+    // Peek the first byte to deduce sign.
+    // Positive values have their sign bit flipped in their encoded format.
+    uint8_t curr_byte = uint8_t(src[0]);
+    const bool src_is_positive = perform_encode ? curr_byte < 0x80 : curr_byte >= 0x80;
+
+    if (src_is_positive) {
+        constexpr uint_t sign_mask = uint_t(1) << (sizeof(T) * 8 - 1);
+        // flip sign bit and write rest of the bits unchanged
+        out.write<uint_t>(read_simple<uint_t>(src) ^ sign_mask);
+    } else {
+        // invert all bytes;
+        out.write<uint_t>(~read_simple<uint_t>(src));
```

Comment:
Better to read the input and write the output outside the if statement to reduce code footprint.